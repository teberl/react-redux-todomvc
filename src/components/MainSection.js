import React from 'react';
import PropTypes from 'prop-types';
import Footer from './Footer';
import VisibleTodoList from '../containers/VisibleTodoList';
import { SHOW_ALL } from '../constants/todoFilters';

MainSection.propTypes = {
  todosCount: PropTypes.number.isRequired,
  completedCount: PropTypes.number.isRequired,
  actions: PropTypes.shape({
    completeAllTodos: PropTypes.func.isRequired,
    clearCompleted: PropTypes.func.isRequired
  })
};

function MainSection(props) {
  const {
    todosCount,
    completedCount,
    actions: { completeAllTodos, clearCompleted },
    match: { params }
  } = props;
  const filter = params.filter || SHOW_ALL;

  return (
    <section className="main">
      {!!todosCount && (
        <span>
          <input
            className="toggle-all"
            type="checkbox"
            checked={completedCount === todosCount}
            readOnly
          />
          <label onClick={completeAllTodos} />
        </span>
      )}
      <VisibleTodoList filter={filter} />
      <Footer
        completedCount={completedCount}
        activeCount={todosCount - completedCount}
        onClearCompleted={clearCompleted}
        filter={filter}
      />
    </section>
  );
}

export default MainSection;
