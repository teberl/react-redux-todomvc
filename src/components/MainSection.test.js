import React from 'react';
import TestRenderer from 'react-test-renderer';
import { createStore, combineReducers } from 'redux';
import { Provider } from 'react-redux';
import { MemoryRouter as Router, Route } from 'react-router-dom';
import MainSection from './MainSection';

test('MainSection should render correct', () => {
  const actions = {
    completeAllTodos: jest.fn(),
    clearCompleted: jest.fn()
  };
  const state = () => [
    {
      text: 'Test one',
      completed: false,
      id: 0
    },
    {
      text: 'Test two',
      completed: true,
      id: 1
    }
  ];
  const testRenderer = TestRenderer.create(
    <Provider store={createStore(combineReducers({ todos: state }))}>
      <Router>
        <Route
          render={props => (
            <MainSection
              todosCount={2}
              completedCount={1}
              actions={actions}
              {...props}
            />
          )}
        />
      </Router>
    </Provider>
  ).toJSON();

  expect(testRenderer).toMatchSnapshot();
});
