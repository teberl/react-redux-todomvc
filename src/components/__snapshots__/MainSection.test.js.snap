// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainSection should render correct 1`] = `
<section
  className="main"
>
  <span>
    <input
      checked={false}
      className="toggle-all"
      readOnly={true}
      type="checkbox"
    />
    <label
      onClick={[MockFunction]}
    />
  </span>
  <ul
    className="todo-list"
  >
    <li
      className=""
    >
      <div
        className="view"
      >
        <input
          checked={false}
          className="toggle"
          onChange={[Function]}
          type="checkbox"
        />
        <label
          onDoubleClick={[Function]}
        >
          Test one
        </label>
        <button
          className="destroy"
          onClick={[Function]}
        />
      </div>
    </li>
    <li
      className="completed"
    >
      <div
        className="view"
      >
        <input
          checked={true}
          className="toggle"
          onChange={[Function]}
          type="checkbox"
        />
        <label
          onDoubleClick={[Function]}
        >
          Test two
        </label>
        <button
          className="destroy"
          onClick={[Function]}
        />
      </div>
    </li>
  </ul>
  <footer
    className="footer"
  >
    <span
      className="todo-count"
    >
      <strong>
        1
      </strong>
       
      item
       left
    </span>
    <ul
      className="filters"
    >
      <li>
        <a
          className="selected"
          href="/"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          All
        </a>
      </li>
      <li>
        <a
          className=""
          href="/show_active"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          Active
        </a>
      </li>
      <li>
        <a
          className=""
          href="/show_completed"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          Completed
        </a>
      </li>
    </ul>
    <button
      className="clear-completed"
      onClick={[MockFunction]}
    >
      Clear completed
    </button>
  </footer>
</section>
`;
